lib/daru/vector.rb

```diff
@@ -238,9 +238,15 @@ def [](*indexes)
           indexes = indexes.map { |e| named_index_for(e) }
         end
 
-        Daru::Vector.new(
-          indexes.map { |loc| @data[@index[loc]] }, 
-          name: @name, index: indexes, dtype: @dtype)
+        begin
+          Daru::Vector.new(
+            indexes.map { |loc| @data[@index[loc]] },
+            name: @name, index: indexes, dtype: @dtype)
+        rescue NoMethodError
+          raise IndexError, "Specified index #{pos.inspect} does not exist."
+        rescue TypeError
+          raise IndexError, "One or more of specified indices #{pos.inspect} do not exist."
```

Comment:
The correct way to get both `:x` and `:y` is `v[:x, :y]`.

So `v[[:x,:y]]` should return a TypeError. Maybe you can handle the type error differently and provide a more verbose output for the same.